### PR TITLE
feat(#minor): Fix Options schema problem happening during compilation and deployment

### PR DESCRIPTION
**Pull request by corerouter**
Some small changes to Options schema in order to compile and deploy related dopex subgraph.

1. Some changes are needed as the errors happening when compiling, deploying and running subgraphs otherwise.
2. Add PCV as the subgraph dashboard will complain it does not comply with generic schema otherwise.
3. Add two extra kinds of fee type to be same with Perp schema.
4. Modify some entry names to be compliant with naming convention.

### CODE REVIEW

**Comment by ishraq8** [comment]
good additions, thank you

File: schema-derivatives-options.graphql [hunk 1]
```diff
@@ -1015,6 +1030,12 @@ type LiquidityPoolHourlySnapshot @entity(immutable: true) {
   " Hourly Deposit trade notional volume occurred in this pool for a specific input token, in USD. The ordering should be the same as the pool's `inputTokens` field. "
   hourlyDepositVolumeByTokenUSD: [BigDecimal!]!
 
+  " All Inflow trade notional volume occurred in this pool for a specific input token, in native amount. The ordering should be the same as the pool's `inputTokens` field. "
+  _cumulativeDepositVolumeByTokenAmount: [BigInt!]!
```
Comment (dmelotik) [inline schema-derivatives-options.graphql:1034]: Why do we want to add this as a helper field? If we add this here there should be a counterpart in `LiquidityPool`

File: schema-derivatives-options.graphql [hunk 2]
```diff
@@ -1245,9 +1278,17 @@ type ActiveAccount @entity {
   id: Bytes!
 }
 
+enum OptionType {
```
Comment (dmelotik) [inline schema-derivatives-options.graphql:1281]: let's place this at the beginning where all the other enums are

**Comment by dmelotik** [comment]
This technically is a #major api change since it will make incompatible changes, but let's keep it as minor since we have not actually implemented this in production yet. Could you also make the proper version change at the top of the file?

File: schema-derivatives-options.graphql [hunk 3]
```diff
@@ -154,6 +160,9 @@ interface Protocol {
   " Current TVL (Total Value Locked) of the entire protocol "
   totalValueLockedUSD: BigDecimal!
 
+  " Current PCV (Protocol Controlled Value). Only relevant for protocols with PCV. "
+  protocolControlledValueUSD: BigDecimal
```
Comment (dmelotik) [inline schema-derivatives-options.graphql:164]: Is there a reason we want this aside from making it work in subgraphs.xyz?

This is the case bc the options schema is not integrated in subgraphs.xyz yet. It is defaults to look at the perps schema. This is being fixed by Michael C though

Comment (corerouter) [reply schema-derivatives-options.graphql:164]: Without protocolControlledValueUSD, subgraphs.xyz complains the schema does not comply with generic schema. Now with protocolControlledValueUSD, subgraphs.xyz looks like it doesn't support options schema yet. Nevertheless, it is added to just make it works with subgraphs.xyz.

File: schema-derivatives-options.graphql (hunk 1 above)
Comment (corerouter) [reply schema-derivatives-options.graphql]: Fixed at 5b75305b341ea53c57e08fee4dab7eb7420db7e9 I may add them by mistake and they don't need in the schema template.

File: schema-derivatives-options.graphql (hunk 2 above)
Comment (corerouter) [reply schema-derivatives-options.graphql]: Changed at 5b75305b341ea53c57e08fee4dab7eb7420db7e9

**Comment by corerouter** [comment]
> This technically is a #major api change since it will make incompatible changes, but let's keep it as minor since we have not actually implemented this in production yet. Could you also make the proper version change at the top of the file?

@dmelotik Changed the schema contents and versions based on reviews . Thanks!

File: schema-derivatives-options.graphql (hunk 3 above)
Comment (dmelotik) [reply schema-derivatives-options.graphql:172]: @ishraq8 do we have a need for PCV in the options schema?

@corerouter What I am getting at is that we should not add it reactively just to make subgraphs.xyz happy. Whatever we decide, Michael C said he would work on adding the options schema into subgraphs.xyz, so we don't have to worry about that.

**Comment by dmelotik** [comment]
@corerouter quick note, minor is the second version (https://semver.org/) so it should be `1.2.0`

File: schema-derivatives-options.graphql (hunk 3 above)
Comment (corerouter) [reply schema-derivatives-options.graphql]: I see. Just restore PCV part to previous versions before I made the changes at 7cee2193120a9b72b80495e89f048f6a5bcc7cb3 @dmelotik @ishraq8

**Comment by corerouter** [comment]
> @corerouter quick note, minor is the second version (https://semver.org/) so it should be `1.2.0`
Made the changes at 7cee2193120a9b72b80495e89f048f6a5bcc7cb3 Thanks!

**Review by jaimehgb — APPROVED** [review]

**Pull request merged**